### DISCUSSION TITLE
edit message fail login

### DISCUSSION THREAD

**Discussion by Waiker** [jcubic/jquery.terminal#909] [Q&A] [answered]
hi! How can i edit message text if user failed authentication.
My authorization function:
```javascript
                login: function(user, password) {
                    this.clear();
                    return fetch('auth.php', {
                        method: 'POST',
                        body: JSON.stringify({user, password}),
                        headers:{
                            'Content-Type': 'application/json'
                        }
                    }).then(res => res.json());
                },
```
How can I edit the error text: "Wrong username or password try again!" ?

**Comment by jcubic** [comment]
You can customize all the strings with the `strings` option:

```javascript
const term = $('body').terminal({}, {
    login(user, passwd) {
        return null
    },
    strings: {
        wrongPasswordTryAgain: 'Bad Passwd'
    }
});
```

**Comment by Waiker** [comment]
Thanks! Where can i read about `strings `option? It's not on the Wiki.

**Reply by jcubic** [reply, marked as answer]
I think that I should list all the strings at API Reference in [translation section](https://terminal.jcubic.pl/api_reference.php#translation). Also, add this to the Wiki. Right now the TypeScript types and the source code are the only documentation. There is a lot of documentation that is missing.

PS: You should ask this below the answer.

**Reply by jcubic** [reply]
I've added the docs to Wiki: [Strings](https://github.com/jcubic/jquery.terminal/wiki/Strings).

**Reply by Waiker** [reply]
Thank you very much!
